src/js/app.js

```diff
@@ -90,12 +84,36 @@ class BoilerplateScene extends React.Component {
       );
     });
 
+    const { cameraDirection } = this.state;
+
+    let pointerBox;
+    if (cameraDirection) {
+      const position = scaleVector(cameraDirection, 4);
+
+      pointerBox = (
+        <Entity
+          physics-body="boundingBox: 1 1 1; mass: 100; velocity: 0 5 0"
+          geometry="primitive: box"
+          position={[position.x, position.y, position.z]}
+          material={{color: 'red'}}>
+        </Entity>
+      );
+    }
+
     return (
       <Scene physics-world="gravity: 0 -9.8 0">
         <a-assets>
         </a-assets>
         <Camera onRotate={(rotation) => {
-            this.spawnBox(rotation);
+            this.setState({
+              cameraDirection: {
+                x: -1 * Math.sin(degToRad(rotation.y % 360)),
+                y: Math.tan(degToRad(rotation.x)),
+                z: -1 * Math.cos(degToRad(rotation.y % 360)),
+              }
+            });
+
+            this.spawnBox(scaleVector(this.state.cameraDirection, 4));
```

Comment:
you probably want to const the cameraDirection object you create above and pass that here rather than read from stale state to spawn a box